add fillIn support for [contenteditable] elements

ro0gr opened the pull request:
resolves #100

ro0gr commented on addon-test-support/fill-in.js:
```diff
@@ -13,8 +14,19 @@ import wait from 'ember-test-helpers/wait';
   let el = getElementWithAssert(selector);
+
+  if (!isFocusable(el)) {
+    throw new Error('Unable to fill element');
```
this is located outside of `run()` in order to have an ability to catch exceptions

ro0gr commented on addon-test-support/fill-in.js:
```diff
@@ -13,8 +14,19 @@ import wait from 'ember-test-helpers/wait';
-  run(() => el.value = text);
+  run(() => {
+    if (el.contentEditable === 'true') {
+      el.textContent = text;
```
seems like `textContent` is more suitable than `innerHTML`?

cibernox replied:
Not sure. What happens if you pass a string containing html to this? `<p><strong>foo</strong>bar</p>`?
Does this treat is as a string or as HTML? 

And the other way around? If you use `innerHTML = "some text"`? Since "some text" is valid html, seems like the second would be more flexible.

ro0gr replied:
> Does this treat is as a string or as HTML?
I don't think so.

I think you are right.  `innerHTML` seems a better choice.

ro0gr commented on addon-test-support/fill-in.js:
```diff
@@ -13,8 +14,19 @@ import wait from 'ember-test-helpers/wait';
 export function fillIn(selector, text) {
   let el = getElementWithAssert(selector);
+
+  if (!isFocusable(el)) {
```
unfortunately it's wrong. 
according to [isFocusable](https://github.com/cibernox/ember-native-dom-helpers/blob/6adbe755c5e0120ab9bfbf49631da75a8ffc3de0/addon-test-support/-private/is-focusable.js#L2) we treat `<a>` and `<link>` as a fillable elements now. 

I think the most straightforward solution is to extract `isContentEditable` and `isFormControl`(or something) helpers and reuse it in here in `fillIn`

ro0gr commented:
I think it's ready to go

ro0gr commented:
hmm.. one of `waitUntil` test failed for `EMBER_TRY_SCENARIO=ember-lts-2.8`. 
I suspect it's not related to this pr. @cibernox can you please restart build?

cibernox commented:
IDK why I forgot about this. Can you verify if it works and rebase?

ro0gr commented:
Sure, I'll do it till the mid of the week

ro0gr commented on addon-test-support/-private/is-focusable.js:
```diff
@@ -1,12 +1,12 @@
 export default function isFocusable(el) {
-  let focusableTags = ['INPUT', 'BUTTON', 'LINK', 'SELECT', 'A', 'TEXTAREA'];
-  let { tagName, type } = el;
+  let focusableTags = ['LINK', 'A'];
```
I'm curious why `LINK` is in the list. Is it somehow focusable?

ro0gr commented:
@cibernox Sorry for delay...

I've rebased and squashed commits. Also test is expanded to check that `fillIn` is able to insert HTML into the `contenteditable` element.

ro0gr commented:
it would be great to trigger CI. Locally tests pass

cibernox commented:
Thanks! I'll release a new version
